cns/service.go

```diff
@@ -68,19 +68,27 @@ func (service *Service) Initialize(config *common.ServiceConfig) error {
 		if err != nil {
 			return err
 		}
-		// Create the listener.
+
 		listener, err := acn.NewListener(u)
 		if err != nil {
 			return err
 		}
+
 		if config.TlsSettings.TLSPort != "" {
 			// listener.URL.Host will always be hostname:port, passed in to CNS via CNS command
 			// else it will default to localhost
 			// extract hostname and override tls port.
 			hostParts := strings.Split(listener.URL.Host, ":")
-			config.TlsSettings.TLSEndpoint = hostParts[0] + ":" + config.TlsSettings.TLSPort
+			tlsAddress := hostParts[0] + ":" + config.TlsSettings.TLSPort
```

Comment:
it would be safer to use [JoinHostPort](https://pkg.go.dev/net#JoinHostPort), but I wish we were using the related net pkg tooling everywhere we deal with this stuff already.